scripts/rebase.py

```diff
@@ -1,116 +1,10 @@
-    edit    = 'edit'    # edit the content
-    squash  = 'squash'  # meld into previous
-    fixup   = 'fixup'   # meld but drop the subject
-    execute = 'exec'    # run a shell command
-    drop    = 'drop'    # remove the commit
-
-
-class Commit(object):
-    """
-    An entry in an interactive git rebase
-    file, containing the action, hash and
-    subject.
-    """
-
-    def __init__(self, action, hash, subject):
-        self.action = action
-        self.hash = hash
-        self.subject = subject
-
-    def __str__(self):
-        return '{} {} {}'.format(self.action, self.hash, self.subject)
-
-def load_commits(rebase_path):
-    """
-    Load the list of commits we need to deal
-    with from a file generated from an inter-
-    active `git rebase`.
-
-    :param rebase_path: path to the rebase file
-    :return: list of commits
-    """
-    commits = []
-    with open(rebase_path) as rebase_file:
-        for line in rebase_file:
-            line = line.strip('\n')
-            # when we first load a rebase file, all
-            # the commits are in the 'pick' state
-            if line.startswith(Action.pick):
-                commits.append(Commit(
-                    action=Action.pick,
-                    hash=line[5:12],
-                    subject=line[13:]
-                ))
-
-    return commits
-
-
-def dump_commits(rebase_path, commits):
-    """
-    Write the list of commits to the rebase
-    file for `git rebase` to operate on.
-
-    :param rebase_path: path to the rebase file
-    :param commits: commits to write
-    """
-    with open(rebase_path, 'w') as rebase_file:
-        for commit in commits:
-            rebase_file.write(str(commit))
-            rebase_file.write('\n')
-
-
-def validate_commits(commits):
-    """
-    Validate that the list of commits we are
-    about to operate on meets our assumptions.
-
-    :param commits: commits to check
-    """
-    tito_index = -1
-    spec_index = -1
-    for index, commit in enumerate(commits):
-        drop_commit = commit.subject.startswith(_DROP_PREFIX)
-        carry_commit = commit.subject.startswith(_CARRY_PREFIX)
-        squash_commit = commit.subject.startswith(_SQUASH_PREFIX)
-        tito_commit = commit.subject.startswith(_TITO_PREFIX)
-
-        if not drop_commit and not carry_commit and not squash_commit and not tito_commit:
-            print('Commit `{}` ({}) invalid!'.format(commit.subject, commit.hash))
-            exit(1)
-
-        if tito_commit:
-            tito_index = index
-
-        if commit.subject == _SPEC_SUBJECT:
-            spec_index = index
-
-    if tito_index < 0:
-        print('No commit from `tito tag` found!')
-        exit(1)
-
-    if spec_index < 0:
-        print('No carry commit for `.spec` file found!')
-        exit(1)
-
-    if tito_index < spec_index:
-        print('Found the `tito tag` commit before the `.spec` file  commit!')
-        exit(1)
-
+import os
+import re
+import git
```

Comment:
Imports of code in a project should ideally be relative so `$PYTHONPATH` can't screw it up